### PR TITLE
ALMS-125, ALMS-131 ApiKeyCredentialsProvider and LocationCredentialsProvider to stop using Amplify

### PR DESCRIPTION
**Pull request by shah279**
- Refactor ApiKeyCredentialsProvider and LocationCredentialsProvider to stop using Amplify
- Refactoring Region classes and returning the proper client
- Test cases added
- Renaming of classes and code optimization

### CODE REVIEW

File: library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package software.amazon.location.auth.data.response
+
+import com.google.gson.annotations.SerializedName
+
+
+data class Credentials(
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt]: move each class to a separate file with the same name, one file = one class

File: library/src/main/java/software/amazon/location/auth/AuthHelper.kt [hunk 2]
```diff
@@ -15,44 +15,62 @@ class AuthHelper(private val context: Context) {
...
+    suspend fun authenticateWithCognitoIdentityPool(
         identityPoolId: String,
-    ): LocationCredentialsProvider = LocationCredentialsProvider(
-        context,
-        identityPoolId,
-        // Get the region from the identity pool id
-        Regions.fromName(identityPoolId.split(":")[0]),
-    )
+    ): LocationCredentialsProvider {
+        return withContext(Dispatchers.Default) {
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/AuthHelper.kt]: if you make network calls, do not use Dispatchers.Default, use Dispatchers.IO

File: library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt [hunk 3]
```diff
@@ -83,13 +79,39 @@ class LocationCredentialsProvider {
...
+     * This function fetches the identity pool ID and region from the key-value store,
+     * uses these to retrieve an identity ID and credentials from AWS Cognito, and then
+     * stores these credentials using a CognitoCredentialsProvider.
+     *
+     * @return True if the credentials were successfully generated and stored, false otherwise.
+     */
+    suspend fun generateCredentials(): Boolean {
+        val identityPoolId = awsKeyValueStore.get(IDENTITY_POOL_ID)
+        val region = awsKeyValueStore.get(REGION)
+        val cognitoCredentialsHttpHelper = CognitoCredentialsClient(region)
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt]: can these values be null here? if so, throw an exception

File: library/src/main/java/software/amazon/location/auth/utils/AwsRequestSigner.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    headers.names()
+        .map { it.trim().lowercase(Locale.ENGLISH) }
+        .sorted()
+        .joinToString(";")
+
+private fun Request.bodyDigest() =
+    hash(bodyAsString()).lowercase(Locale.ENGLISH)
+
+private fun Request.amazonDateHeader() =
+    header("x-amz-date")
```
Comment (olegfilimonov) [inline library/src/main/java/software/amazon/location/auth/utils/AwsRequestSigner.kt]: shouldnt use the already defined constant here?

**Review by olegfilimonov — CHANGES_REQUESTED** [review]

File: library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt (hunk 1 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/data/response/GetCredentialResponse.kt]: Done

File: library/src/main/java/software/amazon/location/auth/AuthHelper.kt (hunk 2 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/AuthHelper.kt]: Done

File: library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt (hunk 3 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt]: Done

File: library/src/main/java/software/amazon/location/auth/utils/AwsRequestSigner.kt (hunk 4 above)
Comment (shah279) [reply library/src/main/java/software/amazon/location/auth/utils/AwsRequestSigner.kt]: Done

**Pull request merged**